src/transaction.rs

```diff
@@ -21,6 +21,22 @@ impl Transaction {
         self.changes.push(TransactionDetails {col, row, old_state, new_state})
     }
 }
+impl Clone for Transaction {
```

Comment:
Can be removed by putting `#[derive(Clone)]` on the `Transaction` and `TransactionDetails`